### ISSUE TITLE
Migration error on managed PostgreSQL database (Oracle Cloud)

### ISSUE THREAD

**Issue by micgerber** [zebbra/data_aggregator#442]
Hello, 

I'm making changes to the Oracle Cloud Instance and after pulling the last docker image :

```
quay.io/zebbra/data_aggregator   latest    e820429ca562   17 hours ago   210MB
```

and [launching the migration](https://github.com/zebbra/data_aggregator/blob/main/docs/deployment.md#roll-out) I get the following error :

```
06:11:12.219 pid=<0.271.0> [info] execute "CREATE OR REPLACE FUNCTION timestamp_from_uuid_v7(_uuid uuid)\nRETURNS TIMESTAMP WITHOUT TIME ZONE\nAS $$\n  SELECT to_timestamp(('x0000' || substr(_uuid::TEXT, 1, 8) || substr(_uuid::TEXT, 10, 4))::BIT(64)::BIGINT::NUMERIC / 1000);\n$$\nLANGUAGE SQL\nIMMUTABLE PARALLEL SAFE STRICT LEAKPROOF;\n"
** (Postgrex.Error) ERROR 42501 (insufficient_privilege) only superuser can define a leakproof function
    (ecto_sql 3.12.0) lib/ecto/adapters/sql.ex:1078: Ecto.Adapters.SQL.raise_sql_call_error/1
    (elixir 1.17.2) lib/enum.ex:1703: Enum."-map/2-lists^map/1-1-"/2
    (ecto_sql 3.12.0) lib/ecto/adapters/sql.ex:1185: Ecto.Adapters.SQL.execute_ddl/4
    (ecto_sql 3.12.0) lib/ecto/migration/runner.ex:348: Ecto.Migration.Runner.log_and_execute_ddl/3
    (elixir 1.17.2) lib/enum.ex:1703: Enum."-map/2-lists^map/1-1-"/2
    (elixir 1.17.2) lib/enum.ex:1703: Enum."-map/2-lists^map/1-1-"/2
    (ecto_sql 3.12.0) lib/ecto/migration/runner.ex:311: Ecto.Migration.Runner.perform_operation/3
```

The error is related to the LEAKPROOF parameter.
The problem is that this parameter can only be used if the PostgreSQL user has the superuser rights.
With managed databases, the main user (postgres) does not have this right :

![image](https://github.com/user-attachments/assets/acf1bf76-db5d-4094-92c9-a36d15683b94)

Would it be possible to remove this parameter from the migration file ([here](https://github.com/zebbra/data_aggregator/blob/136c2e00252e2f95f63d700eeffe95e0c129801d/priv/repo/migrations/20240807160514_install_ash_functions_extension_4.exs#L128) and [here](https://github.com/zebbra/data_aggregator/blob/136c2e00252e2f95f63d700eeffe95e0c129801d/priv/repo/migrations/20240807160514_install_ash_functions_extension_4.exs#L165))?
I don't know what could be the side effects.

I tested the request without this parameter and it worked.

Thank you for your help.

Michaël, info fauna

**Comment by hwuethrich** [comment]
Hi @micgerber 👋  

Once #449 is merged (and the new docker image built), this should be fixed. 

But you have to make sure you have the following extensions installed, before you run the migrations. Otherwise it will try to install them (which also requires superuser privileges):

- uuid-ossp
- citext
- pg_trgm
- btree_gin

The migrations use the following SQL:

```sql
CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
CREATE EXTENSION IF NOT EXISTS "citext";
CREATE EXTENSION IF NOT EXISTS "pg_trgm";
CREATE EXTENSION IF NOT EXISTS "btree_gin";
```

**Comment by hwuethrich** [comment]
@micgerber The new docker image is available (use tag `sha-b762082` or repull `latest`)

**Comment by micgerber** [comment]
Hi @hwuethrich 

Thanks for the fix!
I will pull and test the new docker image.
